backend/src/database.ts

```diff
@@ -1,49 +1,52 @@
 import config from './config';
-import { createPool, PoolConnection } from 'mysql2/promise';
+import { createPool, Pool, PoolConnection } from 'mysql2/promise';
 import logger from './logger';
 import { PoolOptions } from 'mysql2/typings/mysql';
 
 export class DB {
-  static poolConfig = ():PoolOptions => {
-    let poolConfig:PoolOptions = {
+  static connectionsReady: number[] = [];
+  static pool: Pool | null = null;
+
+  static poolConfig = (): PoolOptions => {
+    const poolConfig: PoolOptions = {
```

Comment:
I believe we can move the `const poolConfig` above the class instead of calling the `getPool()` to generate the config object every database query.